src/flows/components/SecretsList.tsx

```diff
@@ -0,0 +1,166 @@
+import React, {
+  FC,
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+  useMemo,
+} from 'react'
+import {connect, ConnectedProps, useDispatch, useSelector} from 'react-redux'
+
+// Components
+import {
+  Button,
+  ComponentColor,
+  ComponentSize,
+  FlexBox,
+  IconFont,
+  JustifyContent,
+} from '@influxdata/clockface'
+import SearchWidget from 'src/shared/components/search_widget/SearchWidget'
+import FilterList from 'src/shared/components/FilterList'
+import InjectSecretOption from 'src/flows/pipes/RawFluxEditor/FluxInjectionOption'
+import {showOverlay, dismissOverlay} from 'src/overlays/actions/overlays'
+
+// Actions & Selectors
+import {getSecrets} from 'src/secrets/actions/thunks'
+import {getAllSecrets} from 'src/resources/selectors'
+
+// Context
+import {PipeContext} from 'src/flows/context/pipe'
+import {EditorContextType} from 'src/flows/context/editor'
+
+// Utils
+import {event} from 'src/cloud/utils/reporting'
+import {Secret} from 'src/types'
+
+const FilterSecrets = FilterList<Secret>()
+
+const SecretToSelect: FC<{
+  secret: Secret
+  onClick: (secret: Secret) => void
+}> = ({secret, onClick}) => {
+  return (
+    <InjectSecretOption
+      option={secret}
+      onClick={onClick}
+      key={secret.id}
+      testID={`select-secret-${secret.id}`}
+    />
+  )
+}
+
+type ReduxProps = ConnectedProps<typeof connector>
+
+interface Props extends ReduxProps {
+  context: EditorContextType
+}
+
+const SecretsList: FC<Props> = ({
+  context: editorContext,
+  onDismissOverlay,
+  onShowOverlay,
+}) => {
+  const [searchTerm, setSearchTerm] = useState<string>('')
+  const dispatch = useDispatch()
+  const secrets = useSelector(getAllSecrets)
+  const {data} = useContext(PipeContext)
+  const {editor, calcInsertPosition, updateText} = editorContext
+  const {queries, activeQuery} = data || {queries: []}
```

Comment:
i'm starting to get the feeling that this component is so tightly coupled with the flux editor (right here we have a hard assumption about the format that the data is being stored in, a format specific to the flux editor), that we should move this component into that panel's definition and not in the shared `src/flows/components` folder

Reply:
agreed. will do.